packages/compass-aggregations/src/modules/pipeline-builder/text-editor-pipeline.ts

```diff
@@ -84,15 +79,33 @@ const reducer: Reducer<TextEditorState> = (state = INITIAL_STATE, action) => {
     action.type === CONFIRM_NEW ||
     action.type === NEW_PIPELINE
   ) {
-    const stageOperators = action.pipeline
-      ? action.pipeline.map(getStageOperator).filter(Boolean) as string[]
-      : state.stageOperators;
+    // On editor switch or reset, reset the parsed pipeline completely
+    const pipeline = action.pipeline ?? [];
+
     return {
       ...state,
       serverError: null,
       previewDocs: null,
       pipelineText: action.pipelineText,
-      stageOperators,
+      pipeline,
+      syntaxErrors: action.syntaxErrors,
+    };
+  }
+
+  if (
+    isAction<EditorValueChangeAction>(
```

Comment:
Moved this to a separate handler because changing the pipeline text shouldn't affect preview fetching state